src/reducers/eye-dropper.js

```diff
@@ -0,0 +1,55 @@
+const ACTIVATE_EYE_DROPPER = 'scratch-paint/eye-dropper/ACTIVATE_COLOR_PICKER';
+const DEACTIVATE_EYE_DROPPER = 'scratch-paint/eye-dropper/DEACTIVATE_COLOR_PICKER';
+
+const initialState = {
```

Comment:
Maybe this state should be part of the mode state?

Reply:
I'd leave that up to you and @paulkaplan if you think it makes sense – moving this to `reducers/modes.js` would mean adding additional functionality to the mode state, since right now, across all modes, it simply changed the name of the mode. Including more stuff in there would be new, and seem to only apply to the eye dropper. If that's cool with you two, then I can do that for sure

Reply:
There are issues with another tool mode being active while the eye dropper is running – this was causing issues with mouse events, which was why we moved this to being a tool, rather than having its own mouse events. 

I'm fine with moving the data that's needed for the eye dropper into the mode itself as a resolution to the issue of it being in two places, but for a reason I can't seem to pin down, just activating the eye dropper as a tool without deactivating another first disables that tool when the eye dropper is then deactivated after it finishes.

Alternatively – I could move the "isActive" logic to just use the `mode`, but keep the eye dropper info logic in the separate reducer. That way, the activation logic is confined to one place, but the data is not cluttering up the `modes` reducer. Thoughts @fsih ?

Reply:
Re: the 2nd paragraph, after you deactivated the eye dropper, it might be that you have to call tool.activate() on the old tool.

(But don't worry about it since I think the other way we were discussing is probably better anyway)